### DISCUSSION TITLE
How to escape "{strings...}"?

### DISCUSSION THREAD

**Discussion by FlatMapIO** [ivanhofer/typesafe-i18n#647] [Q&A] [answered]
My code looks like this: `placeholder: '{"token_id": 0.5}'`, the generator emits the type `RequiredParams<'"token_id"'>`, I have searched the documentation and can't find the escaping instructions

**Comment by ivanhofer** [comment, marked as answer]
Hi @FlatMapIO, there is currently no way to escape characters. `'{"token_id": 0.5}` does not look like a translation for me. What are you trying to achive?

**Reply by FlatMapIO** [reply]
It's not really a translation, I made a mistake when extracting the translation 😅, but if the `{}` happens to be in the translation should we have an escape method

**Reply by ivanhofer** [reply]
In over two years noone really needed it. Until there is no real need for it, I would not add this.

**Reply by vaerlor** [reply]
I want to display a bold string in my translation. Currently, it's easy to add a parameter like `name` (see ConfirmDelete below):

```js
import type { BaseTranslation } from '../i18n-types';

const en: BaseTranslation = {
	Add: 'Add',
	Cancel: 'Cancel',
	ConfirmDelete: 'Are you sure you want to delete {name}?' // this line
};

export default en;
```

However, if I just add the bold tag, I get it outputted as text  `ConfirmDelete: 'Are you sure you want to delete <b>{name}</b>?' `

Is there any way I can escape some basic html in the translations? Maybe with a formatter?

**Reply by vaerlor** [reply]
FYI, I solved the probem by using the bold parameters outside the translation, where the translation is called.
To be honest, it's not really part of the translation so I will understand if you won't implement this :)

**Reply by ivanhofer** [reply]
In the end it is just text. And your framework probably sanitizes it so it does not alter the HTML as it could be a security concern. See `dangerouslySetInnerHTML` for `react` and `@html` for `Svelte`.

**Reply by vaerlor** [reply]
Thanks for replying. I'm using Svelte and I'll stick to my current variant, rather than use `@html`. That seems to open the door to a lot of unwanted stuff.

**Comment by Nicolab** [comment]
Hello,

Since then, is there any way to escape the special character?

Exemple : 

```
"helloWorld": "Hello \{ World \}"
```

Not escaped

**Reply by vaerlor** [reply]
This library is no longer maintained, so nothing changed since then.

**Reply by vaerlor** [reply]
See issue https://github.com/ivanhofer/typesafe-i18n/issues/739

**Reply by Nicolab** [reply]
@sereneowl Thanks, sad :(
